lua/apisix/admin/upstreams.lua

```diff
@@ -28,6 +28,59 @@ local _M = {
 }
 
 
+local function get_chash_key_schema(hash_on)
+    if not hash_on then
+        return nil, "hash_on is nil"
+    end
+
+    if hash_on == "vars" then
+        return core.schema.upstream_hash_vars_schema
+    end
+
+    if hash_on == "header" or hash_on == "cookie" then
+        return core.schema.upstream_hash_header_schema
+    end
+
+    if hash_on == "consumer" then
+        return nil, nil
+    end
+
+    return nil, "invalid hash_on type " .. hash_on
+end
+
+
+local function check_upstream_conf(conf)
+    local ok, err = core.schema.check(core.schema.upstream, conf)
+    if not ok then
+        return false, {error_msg = "invalid configuration: " .. err}
+    end
+
+    if conf.type == "chash" then
```

Comment:
Bad code style.

Here is the good style:

```lua
if conf.type ~= "chash" then
    return true
end

if not conf.hash_on then
    conf.hash_on = "vars"
end

...
return true
```

Reply:
ok,  got it.